crates/compiler/test_gen/src/gen_primitives.rs

```diff
@@ -3278,6 +3294,17 @@ fn box_num() {
     assert_evals_to!("Box.box 123u64", RocBox::new(123), RocBox<u64>)
 }
 
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+#[ignore = "triggers some UB somewhere in at least the llvm and dev backends"]
+fn box_str() {
+    assert_evals_to!(
+        "Box.box \"short\"",
+        RocBox::new(RocStr::from("short")),
+        RocBox<RocStr>
+    );
+}
+
```

Comment:
so, yeah, no idea yet what this is. I believe it's on the roc side, and observe it in at least the llvm and dev backends (but only when running in release mode). That suggests it's on the rust side though because for the dev backend `--release` on the compiler should not matter (that is, it should produce the same assembly either way). 

very confusing, but not specifically a dev backend issue, so I want to tackle that separately.

Reply:
I wonder if rust is optimizing something away due to the extra level of indirection.

That or maybe it is somehow an issue with small string optimization. Though that wouldn't make sense to be affect by release.